Sources/Storage/Disk/Config/DiskConfigurable.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  DiskConfigurable.swift
+//  StatsdClient
+//
+//  Created by Nghia Tran on 10/8/17.
+//  Copyright © 2017 StatsdClient. All rights reserved.
+//
+
+import Foundation
+
+protocol DiskConfigurable {
+
+    // Name
+    // It's also name of folder cache
+    var name: String { get }
+
+    // Max size on bytes
+    var maxSize: UInt { get }
+
+    // Cache folder type
+    var directoryType: DirectoryType { get }
+
+    // Protection mode in iOS/tvOS
+    var protectionType: FileProtectionType? { get }
```

Comment:
I'd say we don't need disk protection for now, this'd add extra unneeded complexity in, as we'd have to handle the differences across platforms.